### ISSUE TITLE
resolve.extensions not working for files different from js, jsx

### ISSUE THREAD

**Issue by DenRedsky** [webpack/webpack#12042]
# Bug report

**What is the current behavior?**
https://webpack.js.org/configuration/resolve/#resolveextensions

```extensions: ['.jsx', '.js', '.styl', '.svg']```

For example, i have folder with ```done.svg, index.jsx, styles.styl``` files.

Importing svg file w/o extension in index.jsx file - **success**:

```
import React from 'react';
import DoneIcon from './icon/done';
import styles from './styles';
```
Importing svg file w/o extension in styles.styl - **failed**:

```
.benefits
  margin-top 16px

  &__list_item
    line-height 24px

    &:before
      content url('./icon/done')
      margin-right 8px
      vertical-align text-top
```
<img width="588" alt="Снимок экрана 2020-11-23 в 20 40 44" src="https://user-images.githubusercontent.com/10866181/99996510-e98f6800-2dcc-11eb-8ccd-00cf4ee32f65.png">

**If the current behavior is a bug, please provide the steps to reproduce.**

```
module.exports = {
  mode: 'production',
  output: {
    path: path.resolve('dist'),
    filename: 'js/[name].[contenthash].js',
    assetModuleFilename: '[name].[contenthash][ext]'
  },
  resolve: {
    extensions: ['.jsx', '.js', '.styl', '.svg']
  },
  devtool: false,
  stats: {
    all: false,
    timings: true,
    assets: true,
    assetsSort: 'size',
    errors: true,
    warnings: true
  },
  module: {
    rules: [
      {
        test: /\.(js|jsx)$/,
        use: 'babel-loader',
        exclude: /node_modules/
      },
      {
        test: /\.styl$/,
        use: [
          MiniCss.loader,
          {
            loader: 'css-loader',
            options: {
              modules: true
            }
          },
          'postcss-loader',
          'stylus-loader'
        ],
        exclude: /node_modules/
      },
      {
        test: /\.(svg|jpg|webp)$/i,
        type: 'asset'
      }
    ]
  },
  optimization: {
    minimizer: ['...', new CssMinimizerPlugin()],
    runtimeChunk: 'single',
    splitChunks: {
      minChunks: 2,
      chunks: 'all'
    }
  },
  plugins: [
    new CleanWebpackPlugin(),
    new HtmlWebpackPlugin({
      template: path.resolve('src', 'index.html'),
      minify: true
    }),
    new MiniCss({ filename: 'css/[name].[contenthash].css' }),
    new CopyWebpackPlugin({
      patterns: [{ from: path.resolve('manifest'), to: path.resolve('dist') }]
    }),
    new CheckDuplicatePlugin()
  ]
};
```

**What is the expected behavior?**
Like in webpack 4, using file with extension from resolve.extensions list is not leading to error.

**Other relevant information:**
webpack version:  5.6.0
Node.js version: 12.16.2
Operating System: MacOS Mojave

**Comment by evilebottnawi** [comment]
I don't understand what you mean:
```
content url('./icon/done') - Error: Can't resolve 'icon/done' in.
```

I don't think it will be work in browser without webpack

**Comment by evilebottnawi** [comment]
Maybe you want:
```
resolve: {
  byDependency: {
    url: {
      preferRelative: true
      extensions: ['.svg']
    },
  }
}
```

**Comment by evilebottnawi** [comment]
Resolving is different for CommonsJS, ESM, URL, CSS and etc

**Comment by DenRedsky** [comment]
> I don't understand what you mean:
> 
> ```
> content url('./icon/done') - Error: Can't resolve 'icon/done' in.
> ```
> 
> I don't think it will be work in browser without webpack

i mean i have style file with styles, where i use **svg** file.

```
.benefits
  $block()
  margin-top 16px

  &__list_item
    line-height 24px

    &:before
      content url('./icon/done')
      margin-right 8px
      vertical-align text-top
```

**Comment by evilebottnawi** [comment]
Do you try https://github.com/webpack/webpack/issues/12042#issuecomment-732270435?

But it will not work without webpack, so you should avoid to using this, this options not for this

**Comment by DenRedsky** [comment]
> Do you try [#12042 (comment)](https://github.com/webpack/webpack/issues/12042#issuecomment-732270435)?
> 
> But it will not work without webpack (try to compile sass to css and run it in browser), so you should avoid to using this, this options not for this

Yes, i try. Not worked for me. Sorry, but i don't understand, what does it mean "not work without webpack"? It's all working under webpack. And it's compiling in css of course.

**Comment by evilebottnawi** [comment]
You should avoid using non-standard syntax as much as possible, if you try to use `url('./icon/done')` in browsers without webpack you will get the problem, because you expected what it will be url to file, but you provide url to directory and browser don't show image. Like ES modules require `.js`/`.mjs` extensions in a browser.

Anyway it should be work right now, but I think here problems with `stylus-loader` or `css-loader`, I don't help you because you don't provide example of code, just config and error, if you want help ASAP please provide as much information as possible

**Comment by DenRedsky** [comment]
> if you try to use `url('./icon/done')` in browsers without webpack you will get the problem

i don't use it w/o webpack, after production build it's compile in base64 link =)

**Comment by evilebottnawi** [comment]
Please provide example of code + config

**Comment by DenRedsky** [comment]
> Please provide example of code + config

i updated first message, may be it will be helpfull. But can't give more info, cuz it's not public code.

**Comment by evilebottnawi** [comment]
Problem on `css-loader` side... Hope I will found time to look at this in near future

**Comment by alexander-akait** [comment]
@webpack-bot move to `webpack-contrib/css-loader`

**Comment by Piemontez** [comment]
I have some problem for similar situation.

In my case, i have one system work in web broswer, and work in dekstop electron app.
The webpack and the system worked perfectly as long as I didn't need to separate functionality from the web and desktop systems.

To solve this situation, I separated files with different extensions webview.tsx and desktop.tsx as "FileOpenMenu.desktop.tsx" and "FileOpenMenu.webview.tsx". And for import I would like to import like this "import FileOpenMenu from 'FileOpenMenu".

For this I needed to create 2 webpack files that would load the correct file depending on the final packaging, as shown below.

webpack.desktop.jsx
extensions: ['**deskopt.jsx**', '.jsx', '.js']

webpack.webview.jsx
extensions: ['**webview.tsx**', '.jsx', '.js']

However webpack does not resolve the "**desktop.tsx**" and "**webview.tsx**" extensions.
